core/base/inc/TString.h

```diff
@@ -511,49 +530,31 @@ inline TString &TString::operator+=(const TString &s)
 inline TString &TString::operator+=(char c)
 { return Append(c); }
 
-inline TString &TString::operator+=(Long_t i)
-{ char s[32]; sprintf(s, "%ld", i); return operator+=(s); }
-
-inline TString &TString::operator+=(ULong_t i)
-{ char s[32]; sprintf(s, "%lu", i); return operator+=(s); }
-
-inline TString &TString::operator+=(Short_t i)
-{ return operator+=((Long_t) i); }
-
-inline TString &TString::operator+=(UShort_t i)
-{ return operator+=((ULong_t) i); }
-
-inline TString &TString::operator+=(Int_t i)
-{ return operator+=((Long_t) i); }
-
-inline TString &TString::operator+=(UInt_t i)
-{ return operator+=((ULong_t) i); }
-
-inline TString &TString::operator+=(Double_t f)
+template<class T>
+inline typename std::enable_if<is_signed_numeral<T>::value,TString>::type
+&TString::operator+=(T i)
 {
-   char s[32];
-   // coverity[secure_coding] Buffer is large enough: width specified in format
-   sprintf(s, "%.17g", f);
-   return operator+=(s);
+   char buffer[32];
+   snprintf(buffer, sizeof(buffer), "%lld", static_cast<Long64_t>(i));
+   return operator+=(buffer);
 }
 
-inline TString &TString::operator+=(Float_t f)
-{ return operator+=((Double_t) f); }
-
-inline TString &TString::operator+=(Long64_t l)
+template<class T>
+inline typename std::enable_if<is_unsigned_numeral<T>::value,TString>::type
+&TString::operator+=(T u)
 {
-   char s[32];
-   // coverity[secure_coding] Buffer is large enough (2^64 = 20 digits).
```

Comment:
Could you keep these comments at the relevant places?

Reply:
Yes, I'll put them back. I didn't mean for them to be erased.